comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -30,16 +39,22 @@ const (
 )
 
 type collector struct {
-	id                  string
-	store               workloadmeta.Component
-	catalog             workloadmeta.AgentType
-	metaV1              v1.Client
-	metaV3or4           func(metaURI, metaVersion string) v3or4.Client
-	clusterName         string
-	hasResourceTags     bool
-	collectResourceTags bool
-	resourceTags        map[string]resourceTags
-	seen                map[workloadmeta.EntityID]struct{}
+	id                      string
+	store                   workloadmeta.Component
+	catalog                 workloadmeta.AgentType
+	metaV1                  v1.Client
+	metaV3or4               func(metaURI, metaVersion string) v3or4.Client
+	clusterName             string
+	hasResourceTags         bool
+	collectResourceTags     bool
+	resourceTags            map[string]resourceTags
+	seen                    map[workloadmeta.EntityID]struct{}
+	v4TaskEnabled           bool
```

Comment:
what does this option means? T4 Task collection? 
Shou we name it instead `TaskCollectionEnabled` because we might have later v5 and v6.